File: BasicOOP/src/object/Kamus.java
/**
 * @author radityo.p.w (radityo.p.w@gmail.com)
 * @version 1.0
 */

package object;

import java.util.HashMap;


public class Kamus {


    HashMap<String,Mata> daftarMata;

    public HashMap<String, Mata> getDaftarMata() {

        if(daftarMata == null){

            daftarMata = new HashMap<String, Mata>();

            Mata m = new Mata();

            m.setIsBeleken(Boolean.TRUE);
            m.setJenisMata("Mata Sakit");
            m.setWarnaBolaMata("Merah");

            daftarMata.put(m.getJenisMata(), m);

            Mata m1 = new Mata();

            m1.setIsBeleken(Boolean.FALSE);
            m1.setJenisMata("Mata Sehat");
            m1.setWarnaBolaMata("Biru");

            daftarMata.put(m1.getJenisMata(), m1);


            Mata m2 = new Mata();
            m2.setIsBeleken(Boolean.FALSE);
            m2.setWarnaBolaMata("IJO");
            m2.setJenisMata("Mata Duitan");

            daftarMata.put(m2.getJenisMata(), m2);



        }


        return daftarMata;
    }


    HashMap<String,KacaMata> daftarKacaMata;

    public HashMap<String, KacaMata> getDaftarKacaMata() {

        if(daftarKacaMata == null){

            daftarKacaMata = new HashMap<String,KacaMata>();

            KacaMata km = new KacaMata();

            km.setWarna("merah");

            km.setMerk("Kaca Mata Kijang");

            km.setHarga("3000");


            daftarKacaMata.put(km.getMerk(), km);


            KacaMata km1 = new KacaMata();

            km1.setWarna("putih");

            km1.setMerk("Kaca Mata Kuda");

            km1.setHarga("40000");

            daftarKacaMata.put(km1.getMerk(), km1);


            KacaMata km2 =  new KacaMata();

            km2.setWarna("hitam");

            km2.setMerk("Kaca Mata Orang");

            km2.setHarga("50000");


        }


        return daftarKacaMata;
    }



       

}

File: BasicOOPRubyBased/lib/object/kamus.rb
# @author radityo.p.w (radityo.p.w@gmail.com)
# @version 1.0

class Kamus

  @daftar_mata = nil

  @daftar_kaca_mata = nil


  def get_daftar_mata

    if @daftar_mata == nil

      @daftar_mata = Hash.new

      m = Mata.new

      m.is_beleken = true
      m.jenis_mata = "Mata Sakit"
      m.warna_bola_mata = "Merah"


      m1 = Mata.new
      m1.is_beleken = false
      m1.jenis_mata = "Mata Sehat"
      m1.warna_bola_mata = "biru"

      m2 = Mata.new
      m2.is_beleken = false
      m2.warna_bola_mata = "IJO"
      m2.jenis_mata = "Mata Duitan"


      @daftar_mata[:mata_sakit] = m
      @daftar_mata[:mata_duitan] = m2
      @daftar_mata[:mata_sehat] = m1

      

    end

    @daftar_mata
    
  end


  def get_daftar_kaca_mata

    if @daftar_kaca_mata == nil


      @daftar_kaca_mata = Hash.new

      km = KacaMata.new

      km.warna = "merah"

      km.merk = "Kaca Mata Kijang"

      km.harga = "3000"



      km1 = KacaMata.new

      km1.warna = "putih"

      km1.merk = "Kaca Mata Kuda"

      km1.harga = "40000"


      km2 = KacaMata.new

      km2.warna = "hitam"

      km2.merk = "Kaca Mata Orang"

      km2.harga = "50000"


      @daftar_kaca_mata[:kaca_mata_orang] = km2

      @daftar_kaca_mata[:kaca_mata_kuda] = km1

      @daftar_kaca_mata[:kaca_mata_kijang] = km


    end


    @daftar_kaca_mata
    

  end


end
